### PR TITLE
60832_create_subject_with_structure

### PR DESCRIPTION
**Pull request by ahammad-reyad**
Created subject with structure

### CODE REVIEW

File: app/controllers/admin/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Admin::SubjectsController < ApplicationController
+  def new
```
Comment (MuhammadTamzid) [inline app/controllers/admin/subjects_controller.rb]: please use cancancan :)

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -7,4 +7,19 @@ def full_title page_title = ""
       page_title + " | " + base_title
     end
   end
+
+  def link_to_add_fields name, f, association, cssClass, title
```
Comment (MuhammadTamzid) [inline app/helpers/application_helper.rb]: we can do this in one line using "nested_form" gem :D

File: app/models/subject.rb [hunk 3]
```diff
@@ -1,5 +1,11 @@
 class Subject < ActiveRecord::Base
-  has_many :task, dependent: :destroy
+  has_many :tasks, dependent: :destroy
   has_many :course_subjects, dependent: :destroy
   has_many :courses, through: :course_subjects
+
+  accepts_nested_attributes_for :tasks, reject_if:
+   lambda {|attribute| attribute[:title].blank?}, allow_destroy: true
+
+   validates :title, presence: true
```
Comment (MuhammadTamzid) [inline app/models/subject.rb]: indent

File: app/models/subject.rb [hunk 4]
```diff
@@ -1,5 +1,11 @@
...
+   validates :instruction, presence: true
```
Comment (MuhammadTamzid) [inline app/models/subject.rb]: indent

File: app/controllers/admin/subjects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+class Admin::SubjectsController < ApplicationController
+  load_and_authorize_resource
+  def new
```
Comment (MuhammadTamzid) [inline app/controllers/admin/subjects_controller.rb]: add a new line for separate

File: app/controllers/admin/subjects_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    @subject = Subject.new
```
Comment (MuhammadTamzid) [inline app/controllers/admin/subjects_controller.rb]: used ``load_and_authorize_resource`` so don't need this line

File: app/controllers/admin/subjects_controller.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    @subject.tasks.build
+  end
+
+  def create
+    @subject = Subject.new subject_params
```
Comment (MuhammadTamzid) [inline app/controllers/admin/subjects_controller.rb]: don't need

File: config/routes.rb [hunk 8]
```diff
@@ -6,4 +6,7 @@
     sessions: "users/sessions",
     registrations: "users/registrations"
   }
+  namespace :admin do
+    resources :subjects
```
Comment (dieunb) [inline config/routes.rb]: pls, add admin root page

**Pull request merged**